autoprotocol/protocol.py

```diff
@@ -4979,7 +4973,7 @@ def provision(self, resource_id, dests, volumes):
                     d.container):
                 self.instructions[-1].to.append(xfer)
             else:
-                return self._append_and_return(Provision(resource_id,
+                self._append_and_return(Provision(resource_id,
```

Comment:
hmm.. was going to write we should keep this, but I can see how this introduced a bug.

Basically the return here could potentially short-circuit the for-loop, which will fail in the case there are multiple wells.

This seems similar to the `transfer` case, where it could generate `n` LiquidHandle Instructions. Let's return a list here, similar to how its done there.

I.e. accumulate to some list here in this statement and return that list at the end. Let's change the type signature of the return above as well to `list(Provision)`

Reply:
Returning `list(Provision)` also doesn't makes sense since we are adding additional instructions(`uncover` & `unseal`), conditionally. Anyhow I made changes to capture & return only `Provision` instructions to retrofit other existing methods implementation pattern. 

Generation observation: I see that `Protocol` class is following builder design pattern(https://refactoring.guru/design-patterns/builder/python/example) , where it is enhancing the protocol object. But at the same time, we are expecting each method to return only what is being added(which is more of like a utility method where we pass certain params and it returns some object)